### PR TITLE
📣 Refactor of announce entry validation

### LINKED ISSUE #2777: AnnounceWorkEntry - validation refactor
Context
Goal of this issue is to clean up the validation logic on Bounty modals to help further development/bug resolving for any developer involved

Scope
Purpose of refactor is to remove doubled validation on each modal and stick only to the Yup one. For UX/UI side everything should be the same, this issue is some way is a pre-cursor to the https://github.com/Joystream/pioneer/issues/2802 and is about improving speed and ease of future community development.

Corresponding PR: https://github.com/Joystream/pioneer/pull/2778

### PR DESCRIPTION
**Pull request by WRadoslaw**
Close #2777

### CODE REVIEW

File: packages/ui/src/bounty/modals/AnnounceWorkEntryModal/machine.ts [hunk 1]
```diff
@@ -9,10 +10,16 @@ import {
...
-interface TransactionContext {
+interface ContributionContext {
+  stakingAccount: Account
+}
+
+interface TransactionContext extends ContributionContext {
   transactionEvents?: EventRecord[]
 }
 
+type AnnounceWorkEntryContext = Partial<ContributionContext & TransactionContext>
```
Comment (thesan) [inline packages/ui/src/bounty/modals/AnnounceWorkEntryModal/machine.ts:21]: AFAICS one of these types is redundant. Why not:
```suggestion
interface ContributionContext {
  stakingAccount?: Account
}

interface TransactionContext extends ContributionContext {
  transactionEvents?: EventRecord[]
}
```
And later:
```ts
  | { value: AnnounceWorkEntryStates.contribute; context: Reqired<ContributionContext> }
```

**Review by thesan — CHANGES_REQUESTED** [review]
Looks good ! There's a detail about the context types, but mostly it looks like `@/memberships/model/validation` has duplicated logic. Did you rename this file in  #2775 ?

**Comment by WRadoslaw** [comment]
> Looks good ! There's a detail about the context types, but mostly it looks like `@/memberships/model/validation` has duplicated logic. Did you rename this file in #2775 ?

Yeah, seems like merge/rebase issue

**Review by thesan — APPROVED** [review]
LGTM !

**Pull request merged**